### ISSUE TITLE
Wifi issues with bettercap

### ISSUE THREAD

**Issue by deanfourie1** [bettercap/bettercap#919]
Latest version of bettercap.

Im getting issues with anything wifi related. I am able to set tx power using iwconfig and also scan and channel hop with airmon and airodump-ng just fine.

Any ideas? As much info as possible provided below.

bettercap -iface wlan0
```
└─# bettercap -iface wlan0   
bettercap v2.32.0 (built for linux amd64 with go1.15.15) [type 'help' for a list of commands]
wlan0  » wifi.recon on
[03:39:50] [sys.log] [inf] wifi using interface wlan0 (a6:c4:fe:d1:f7:3f)
[03:39:50] [sys.log] [war] wifi could not set interface wlan0 txpower to 30, 'Set Tx Power' requests not supported
[03:39:50] [sys.log] [war] wifi error while activating handle: error while setting interface wlan0 in monitor mode: Cannot set rfmon for this handle, interface might already be monitoring. retrying!
[03:39:50] [sys.log] [inf] wifi started (min rssi: -200 dBm)
 wlan0  » [03:39:50] [sys.log] [inf] wifi channel hopper started.
 wlan0  » [03:40:21] [sys.log] [war] wifi error while hopping to channel 169: iw: out=kernel reports: Channel is disabled
command failed: Invalid argument (-22) err=exit status 234
 wlan0  » [03:40:21] [sys.log] [war] wifi error while hopping to channel 173: iw: out=kernel reports: Channel is disabled
command failed: Invalid argument (-22) err=exit status 234
 wlan0  »  
```

uname -r
```
└─# uname -r
5.10.0-kali9-amd64
```

lsusb
`Bus 002 Device 002: ID 0e8d:7612 MediaTek Inc. MT7612U 802.11a/b/g/n/ac Wireless Adapter`

lsusb -v
```
Bus 002 Device 002: ID 0e8d:7612 MediaTek Inc. MT7612U 802.11a/b/g/n/ac Wireless Adapter
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               3.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0         9
  idVendor           0x0e8d MediaTek Inc.
  idProduct          0x7612 MT7612U 802.11a/b/g/n/ac Wireless Adapter
  bcdDevice            1.00
  iManufacturer           2 MediaTek Inc.
  iProduct                3 Wireless 
  iSerial                 4 000000000
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x007a
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          5 BT_Config
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              400mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           8
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass    255 Vendor Specific Subclass
      bInterfaceProtocol    255 Vendor Specific Protocol
      iInterface              1 WiFi_If
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x85  EP 5 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x08  EP 8 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x04  EP 4 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x05  EP 5 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x06  EP 6 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x07  EP 7 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x09  EP 9 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0400  1x 1024 bytes
        bInterval               0
        bMaxBurst               1
Binary Object Store Descriptor:
  bLength                 5
  bDescriptorType        15
  wTotalLength       0x0016
  bNumDeviceCaps          2
  USB 2.0 Extension Device Capability:
    bLength                 7
    bDescriptorType        16
    bDevCapabilityType      2
    bmAttributes   0x0000f41e
      BESL Link Power Management (LPM) Supported
    BESL value     1024 us 
    Deep BESL value    61440 us 
  SuperSpeed USB Device Capability:
    bLength                10
    bDescriptorType        16
    bDevCapabilityType      3
    bmAttributes         0x00
    wSpeedsSupported   0x000e
      Device can operate at Full Speed (12Mbps)
      Device can operate at High Speed (480Mbps)
      Device can operate at SuperSpeed (5Gbps)
    bFunctionalitySupport   1
      Lowest fully-functional device speed is Full Speed (12Mbps)
    bU1DevExitLat          10 micro seconds
    bU2DevExitLat         180 micro seconds
Device Status:     0x000c
  (Bus Powered)
  U1 Enabled
  U2 Enabled
```

**Comment by palulabrasil** [comment]
I'm having a similar problem regarding this issue. The result just isn't as informatively thorough as @deanfourie1's is...

bettercap -iface wlp3s0
```
└─# wifi.recon on
[20:07:10] [sys.log] [inf] wifi using interface wlp3s0 (redacted)
[20:07:10] [sys.log] [war] wifi could not set interface wlp3s0 txpower to 30, 'Set Tx Power' requests not supported
 wlp3s0  » [20:07:10] [sys.log] [err] error while setting interface wlp3s0 in monitor mode: Cannot set rfmon for this handle
 wlp3s0
```
uname -r
```
└─# uname -r
5.15.0-69-generic
```

**Comment by evilsocket** [comment]
fixed by 2.33.0
